### PR TITLE
[Docs] Ensure breaking-changes tag exists for empty and existing breaking changes

### LINKED ISSUE #94910: [DOCS] Release notes generation removes notable-breaking-changes tag
It seems the release notes generation automation erroneously removes the `notable-breaking-changes` tag from the migration guide, see for example [here](https://github.com/elastic/elasticsearch/pull/93653/files#diff-053ec541f7e0c214a1a42ebc54233fa7e5bda9a36816e008d6463dc162b57a85L19). Ideally, this gets fixed before the next minor release.

### PR DESCRIPTION
**Pull request by craigtaverner**
The original feature only added the tags when there were breaking changes with `notable: true`. Last year we improved this by adding the tag when there were no breaking changes (in 8.3.0). However we missed the case where there are breaking changes, but the `notable: false` means the tags did not get added. This fixes that case.

I manually tested this with three cases:

* No breaking changes
* Only one notable=false
* One notable:false and one notable:true

Fixes #94910

### CODE REVIEW

File: build-tools-internal/src/main/resources/templates/breaking-changes.asciidoc [hunk 1]
```diff
@@ -15,10 +15,10 @@ coming::[${majorDotMinorDotRevision}]
 [discrete]
 [[breaking-changes-${majorDotMinor}]]
 === Breaking changes
-<% if (breakingByNotabilityByArea.isEmpty()) { %>
+
 // tag::notable-breaking-changes[]
```
Comment (abdonpijpelink) [inline build-tools-internal/src/main/resources/templates/breaking-changes.asciidoc:19]: If there are breaking changes, the opening `// tag::notable-breaking-changes[]` tag should be lower, on line 27 (after the "The following changes.." paragraph. See for example: https://github.com/elastic/elasticsearch/blob/main/docs/reference/migration/migrate_8_5.asciidoc?plain=1#L22

**Review by abdonpijpelink — COMMENTED** [review]
I've left a comment

File: build-tools-internal/src/main/resources/templates/breaking-changes.asciidoc (hunk 1 above)
Comment (craigtaverner) [reply build-tools-internal/src/main/resources/templates/breaking-changes.asciidoc:19]: Looking further down I see there is already a notable-breaking-changes tag. So this fix is not valid. The alternative explanation is that there are no breaking changes with the `notable` flag. We need an alternative fix.

**Review by abdonpijpelink — APPROVED** [review]
LGTM. Thanks @craigtaverner !

**Pull request merged**